catalog/core/catalog-core-api/src/main/java/ddf/catalog/configuration/SearchCapabilityConfiguration.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.configuration;
+
+/**
+ * Interface to determine what search capabilities are enabled.
```

Comment:
I struggled with this for quite some time.  I originally had this as part of the provider instead because I agree that it makes more sense there.  However, if we want to support highlighting data from the cache I needed it somewhere else because having the configuration in the provider and used by the cache creates a dependency problem.  If the cache were not embedded inside the catalog framework, I would have done this differently.  I'm open to suggestions.

Reply:
Perhaps the best answer is we don't support it in the cache right now and then add in that capability once the cache is extracted out of the catalog framework?

Reply:
The other reason this is here (now that I'm going through this again) is that the actual solr client code that needs this is in `catalog-core-solr`, not in the provider, and that code is utilized by both the solr provider and the framework.  I needed a common place to put the API that didn't cause dependency issues.  So, even if we don't enable this for the cache, I still need a solution to the dependency problem if it is not here in the api.

Reply:
I like the idea of not supporting it in the cache for now.  Maybe we can get the cache extracted soon. But even if we did support it in the cache, I think the cache or at least the caching federation strategy has its own configuration we could add this to.